mindmeld/resource_loader.py

```diff
@@ -417,6 +417,38 @@ def _traverse_labeled_queries_files(
                     if re.match(file_pattern, os.path.basename(file_path)):
                         yield a_domain, an_intent, file_path
 
+    def get_all_file_paths(self):
+        """ Get a list of text file paths across all intents.
+
+        Returns:
+            list: A list of all file paths.
+        """
+        file_iter = self._traverse_labeled_queries_files(file_pattern=".*.txt")
+        return [filename for _, _, filename in file_iter]
+
+    def filter_file_paths(self, file_pattern=".*", file_paths=None):
+        """ Get a list of file paths that match a specific file_pattern
+
+        Args:
+            file_pattern (str): A regex pattern to filter on.
+            file_paths (list): A list of file paths.
+
+        Returns:
+            list: A list of file paths.
+        """
+        all_file_paths = file_paths or self.get_all_file_paths()
+        compiled_pattern = re.compile(file_pattern)
```

Comment:
Do the compilation on the caller of this function

Reply:
To clarify, are you saying to replace the function parameter 'file_pattern' with 'compiled_pattern'? This way the caller is responsible for running re.compile() and passing the resulting value into this func.